classes/Leth/IPAddress/IP/NetworkAddress.php

```diff
@@ -61,73 +63,85 @@ public static function generate_subnet_mask($cidr)
 	 * @return IP\Address An IP Address representing the mask.
 	 * @author Marcus Cobden
 	 */
-	public static function get_global_netmask()
-	{
+	public static function get_global_netmask(): Address
+    {
 		throw new \LogicException(__METHOD__.' not implemented in subclass of '.__CLASS__);
 	}
 
-	/**
-	 * Creates an IP\NetworkAddress for the supplied string
-	 *
-	 * @param string $address IP Network Address string.
-	 * @param string $cidr Optional CIDR number. If not supplied It is assumed to be part of the address string
-	 * @return IP\NetworkAddress
-	 */
-	public static function factory($address, $cidr = NULL)
-	{
-		if ($address instanceof IP\NetworkAddress)
+    /**
+     * Creates an IP\NetworkAddress for the supplied string
+     *
+     * @param \Leth\IPAddress\IP\NetworkAddress|\Leth\IPAddress\IP\Address|string $address IP Network Address string.
+     * @param int|string|null                                                     $cidr    Optional CIDR number. If not supplied It is assumed to be part of the address string
+     *
+     * @return \Leth\IPAddress\IPv4\NetworkAddress|\Leth\IPAddress\IP\NetworkAddress|\Leth\IPAddress\IPv6\NetworkAddress
+     */
+	public static function factory(NetworkAddress|Address|string $address, int|string|null $cidr = NULL): IPv4\NetworkAddress|NetworkAddress|IPv6\NetworkAddress
```

Comment:
a) the indentation
b) Just musing, but would this be an opportunity to require that $cidr be an int if present? For both v4 and v6 it'll always be in the bounds of 0-128, so there's no real reason to allow a string, and it would save validation and an (int) cast below. This PR is already a BC break and new major release, so this might be the time.

Reply:
Okay ignore my comment regarding the parameter types, that can/should be a different PR.

Please do fix the indentation, though.

Reply:
The same situation. What do you think about using PHPFixer?